cdap-watchdog/src/main/java/co/cask/cdap/logging/write/FileMetaDataManager.java

```diff
@@ -264,6 +271,13 @@ public String call() throws Exception {
                   LOG.debug("Got file {} with start time {}", file, Bytes.toLong(colName));
                 }
 
+                if (Strings.isNullOrEmpty(namespacedLogDir)) {
+                  LOG.warn("Delete metadata for {} namespace", namespaceId.getEntityName());
+                  table.delete(rowKey, colName);
+                  deletedColumns++;
+                  continue;
+                }
+
                 Location fileLocation = impersonator.doAs(namespaceId, new Callable<Location>() {
```

Comment:
oh I see. That is, I did not see that. That happens when viewing diffs without full context :) 
This is what we want: log a warning but keep the meta data.